plugins/admin-settings/email-types/index.jsx

```diff
@@ -0,0 +1,164 @@
+import { useEffect, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import {
+  Button, SelectControl, TextControl, withNotices,
+} from '@wordpress/components';
+import ImagePicker from '@/components/imagePicker';
+import {
+  Checkboxes, Sortable, SortableItem, useOption,
+} from '@alleyinteractive/block-editor-tools';
+import apiFetch from '@wordpress/api-fetch';
+import { v4 as uuidv4 } from 'uuid';
+
+const AdminEmailTypes = withNotices(
+  ({ noticeOperations, noticeUI }) => {
+    const {
+      value: emailTypes = {},
+      isEdited = false,
+      isSaving = false,
+      onChange = () => {},
+      onSave = () => {},
+    } = useOption('nb_email_types');
+
+    const {
+      value: settings = {},
+    } = useOption('nb_settings');
+
+    const {
+      from_names: fromNames = [],
+    } = settings;
+
+    const fromNameOptions = fromNames.map(
+      (name) => ({ label: name, value: name }),
+    );
+
+    // Get all posts with the nb_template post type.
+    const [templatePosts, setTemplatePosts] = useState([]);
+    useEffect(() => {
+      apiFetch({ path: '/wp/v2/nb_template' })
+        .then((response) => {
+          const templateOptions = response.map(
+            (template) => {
+              const {
+                id: value = 0,
+                title: {
+                  rendered: label = '',
+                } = {},
+              } = template;
+              return { label, value };
+            },
+          );
+          setTemplatePosts(templateOptions);
+        });
+    }, []);
+
+    /**
+     * Save settings data and display a notice.
+     */
+    const { createErrorNotice, createNotice, removeAllNotices } = noticeOperations;
+    const saveSettingsData = () => {
+      onSave().then(() => {
```

Comment:
Generally I prefer the `async`/`await` syntax vs. promises (it eliminates additional nesting). You can wrap an `await` call in `try`/`catch` to get the same behavior of `.catch()` and `.finally()` and so on.